src/plugins/initiatives/initiativesDirectory.js

```diff
@@ -61,3 +70,31 @@ export function fromJSON(j: Compatible<any>): InitiativeFile {
 export function toJSON(m: InitiativeFile): Compatible<InitiativeFile> {
   return toCompat(COMPAT_INFO, m);
 }
+
+/**
+ * When provided with the initiative NodeAddressT of an InitiativeFile this extracts
+ * the URL from it. Or null when the address is not for an InitiativeFile.
+ */
+export function initiativeFileURL(address: NodeAddressT): string | null {
+  const initiativeFilePrefix = NodeAddress.append(
+    initiativeNodeType.prefix,
+    INITIATIVE_FILE_SUBTYPE
+  );
+
+  if (!NodeAddress.hasPrefix(address, initiativeFilePrefix)) {
+    return null;
+  }
+
+  const parts = NodeAddress.toParts(address);
+  const remoteUrl = parts[4];
+  const fileName = parts[5];
+  return `${remoteUrl}/${fileName}`;
+}
+
+// Creates the InitiativeId for an InitiativeFile.
+export function _initiativeFileId(
+  remoteUrl: string,
```

Comment:
nit: might be nice to take `InitiativeDirectory` object as the first argument, for more type safety / clarity

Reply:
No strong preference here. Done!